File: src/Core/Houston.Application/ViewModel/ConnectorFunctionViewModels/ConnectorFunctionViewModel.cs
namespace Houston.Application.ViewModel.ConnectorFunctionViewModels {
	public class ConnectorFunctionViewModel {
		public Guid Id { get; set; }

		public string Name { get; set; } = null!;

		public string? Description { get; set; }

		public bool Active { get; set; }

		public List<ConnectorFunctionHistorySummaryViewModel> Versions { get; set; } = new List<ConnectorFunctionHistorySummaryViewModel>();

		public Guid ConnectorId { get; set; }

		public string CreatedBy { get; set; } = null!;

		public DateTime CreationDate { get; set; }

		public string UpdatedBy { get; set; } = null!;

		public DateTime LastUpdate { get; set; }
	}
}

File: src/Web/WebAngular/src/app/domain/view-models/connector-function.view-model.ts
import { ConnectorFunctionHistorySummaryViewModel } from "./connector-function-history-summary.view-model";

export interface ConnectorFunctionViewModel {
	id: string;
	name: string;
	description: string | null;
	active: boolean;
	versions: ConnectorFunctionHistorySummaryViewModel[];
	connectorId: string;
	createdBy: string;
	creationDate: string;
	updatedBy: string;
	lastUpdate: string;
}
